spec/models/offer_spec.rb

```diff
@@ -9,5 +9,59 @@
 
       expect(offer.conditions.map(&:text)).to contain_exactly('Provide evidence of degree qualification', 'Do a backflip and send us a video')
     end
+
+    describe '#course_option' do
+      it 'returns the application choice current_course_option' do
+        application_choice = create(:application_choice, current_course_option: create(:course_option))
+        offer = create(:offer, application_choice: application_choice)
+
+        expect(offer.course_option).to eq(application_choice.reload.current_course_option)
+      end
+    end
+  end
+
+  describe '#unconditional' do
+    it 'returns true when there are no conditions' do
+      offer = create(:unconditional_offer)
+
+      expect(offer.unconditional?).to be true
+    end
+  end
+
+  context 'delegators' do
```

Comment:
Similarly, do we need to test delegations?

Reply:
Similar to my comment above, I think yes as the spec also acts as documentation for the model